### ISSUE TITLE
LSTM recurrent dropout doesn't appear to be listening to the learning phase

### ISSUE THREAD

**Issue by cjnolet** [keras-team/keras#8907]
I am running an LSTM encoder/decoder framework as specified in the paper "Deep and confident prediction for time series at Uber". I'm having trouble getting the Monte Carlo dropout to work. I have reached out to @yaringal (author of the paper "A theoretically grounded application of Dropout in Recurrent Neural networks") to make sure I'm doing this properly and he has informed me that he helped implement recurrent dropout in Keras (I believe both for Theano and for Tensorflow).

So, my code looks like this (hand typing from a proprietary network so please forgive my typos, the code executes):
```python
dropout_monte_carlo = K.function([model.layers[0].input, K.learning_phase()], [model.layers[3].output])
layer_output = dropout_monte_carlo([[X], 1])[0]
```

My model looks like this:

```python
model = Sequential()
model.add(LSTM(128, activation = 'tanh', input_shape = (1, 28), implementation = 2, dropout = 0.25, recurrent_dropout = 0.25, return_sequences = True))
model.add(LSTM(32, activation = 'tanh', return_sequences = True, implementation = 2, recurrent_dropout = 0.25, dropout = 0.25))
model.add(LSTM(128, activation = 'tanh', return_sequences = True, implementation = 2, recurrent_dropout = 0.25, dropout = 0.25))
model.add(TimeDistributed(Dense(28))

model.compile(loss = 'mse', optimizer = 'adam')
```

The problem is that invoking dropout_monte_carlo() never changes the output. I would expect that the dropout would have given me slightly varying output such that I can do my uncertainty calculation. 

I have tried both Theano and Tensorflow backends (though my understanding is that the recurrent_dropout has been dropped in the current Theano backend).

**Comment by yaringal** [comment]
Using your `dropout_monte_carlo` function and the Keras sentiment analysis example, I get new realisations (different results) every time I call
```dropout_monte_carlo([x_test[:2], 1])```
-- as expected (note the square brackets around x_test and 1). I'm using Keras version 2.0.8.
Y

**Comment by cjnolet** [comment]
I just realized I'm on 2.0.2 on my dev network. I will try 2.0.8 (or 2.1.2). Thank you for trying this!

Also, You are correct, I was invoking the function in the brackets as it should have been but I did not properly hand-type it into Github. Updating the question to reflect.

**Comment by cjnolet** [comment]
Just installed Keras 2.1.2 and it works as intended. 

Thanks again for your help, Yarin.
